components/server/src/server.ts

```diff
@@ -143,6 +143,17 @@ export class Server<C extends GitpodClient, S extends GitpodServer> {
                     log.warn("Websocket connection attempt with non-matching Origin header: " + info.origin)
                     return callback(false, 403);
                 }
+                if (info.req.url === '/v1') {
+                    try {
+                        await this.bearerAuth.auth(info.req as express.Request)
```

Comment:
Why are we doing authentication in a CSRF guard?

Reply:
I will refactor it.

I moved it to the ws handshake to avoid opening a websocket at all for invalid token and propagate unauthorited error properly clients. Before we will establish the websocket and then close it without rejecting the handshake in the first place.